cmd/soroban-rpc/internal/db/migration.go

```diff
@@ -165,55 +181,69 @@ func (g *guardedMigration) Commit(ctx context.Context) error {
 	return setMetaBool(ctx, g.db, g.guardMetaKey, true)
 }
 
-func GetMigrationLedgerRange(ctx context.Context, db *DB, retentionWindow uint32) (*LedgerSeqRange, error) {
+func GetMigrationLedgerRange(ctx context.Context, db *DB, retentionWindow uint32) (LedgerSeqRange, error) {
 	firstLedgerToMigrate := firstLedger
 	latestLedger, err := NewLedgerEntryReader(db).GetLatestLedgerSequence(ctx)
 	if err != nil && !errors.Is(err, ErrEmptyDB) {
-		return nil, fmt.Errorf("failed to get latest ledger sequence: %w", err)
+		return LedgerSeqRange{}, fmt.Errorf("failed to get latest ledger sequence: %w", err)
 	}
 	if latestLedger > retentionWindow {
 		firstLedgerToMigrate = latestLedger - retentionWindow
 	}
-	return &LedgerSeqRange{
+	return LedgerSeqRange{
 		First: firstLedgerToMigrate,
 		Last:  latestLedger,
 	}, nil
 }
 
-func BuildMigrations(ctx context.Context, logger *log.Entry, db *DB, networkPassphrase string,
-	ledgerSeqRange *LedgerSeqRange,
-) (MultiMigration, error) {
+func BuildMigrations(
+	ctx context.Context, logger *log.Entry, db *DB, networkPassphrase string,
+	ledgerSeqRange LedgerSeqRange,
+) (MultiMigration, LedgerSeqRange, error) {
```

Comment:
I don't think it's necessary to return the applicableRange because MultiMigration already has a `ApplicableRange()` function which calculates the applicable range in the same way you are doing here. So, if we get rid of the applicableRange return value here we can also get rid of `MergeInPlace()` because that's the only place where it is used

Reply:
Yeah, you're right! This was the way I originally approached it but I kept getting an infinite loop for some reason. I tried it again and it worked as intended in [a5bfc3b](https://github.com/stellar/soroban-rpc/pull/282/commits/a5bfc3b4d4270396a036850daea6226d421320ec) :+1: